Add logging and reconnect options to createConnection

edwardmcleodjones opened the pull request:
`HubConnectionBuilder` can take additional chained calls to add logging and automatic reconnect options: https://docs.microsoft.com/en-us/javascript/api/@microsoft/signalr/hubconnectionbuilder?view=signalr-js-latest

This change proposes adding two further optional parameters to the createConnection function to allow these to be passed in where required.

edwardmcleodjones committed:
Add logging and reconnect options to createConnection

`HubConnectionBuilder` can take additional chained calls to add logging and automatic reconnect options: https://docs.microsoft.com/en-us/javascript/api/@microsoft/signalr/hubconnectionbuilder?view=signalr-js-latest

This change proposes adding two further optional parameters to the createConnection function to allow these to be passed in where required.

known-as-bmf commented:
Thank you for your contribution!

I was thinking of doing something similar, but in a different way. What do you think of this:

```ts
const connection = createConnection('my-url', options, builder => {
  // here I can further configure the builder the way I want
  return builder.configureLogging(/** logging stuff... **/)
    .withAutomaticReconnect();
});
```

edwardmcleodjones commented:
Yours looks like a cleaner option!  :)  Then the `useSignalr` function would just take the connection and build it I presume?  Looks good to me.

edwardmcleodjones committed:
Incorporates suggestions by @known-as-bmf

Incorporates suggestions by @known-as-bmf in: https://github.com/known-as-bmf/react-signalr/pull/7#issuecomment-810160789

edwardmcleodjones committed:
Adds optional builder function

Adds optional builder function to apply additional options such as logging/automatic reconnection to the HubConnectionBuilder

edwardmcleodjones commented:
@known-as-bmf Have incorporated your suggestion if you want to take a look at the updated version.

known-as-bmf commented:
Please take a look at #9 

Made it so the delegate takes a builder in and returns a builder. I then do `withUrl` to override what may have been set by the developer to prevent weird behaviors.
